api/v1alpha1/tinkerbell_type.go

```diff
@@ -0,0 +1,131 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:path=tinkerbell,scope=Namespaced,categories=tinkerbell,singular=tinkerbell,shortName=tb
+// +kubebuilder:storageversion
+
+// Tinkerbell represents the tinkerbell stack that is being deployed in the kubernetes where the operator is deployed.
+// Tinkerbell operator watches for different resources such as deployment, services, serviceAccounts, etc. One of those
+// CRs is Tinkerbell which the operator will install the tink-stack based on its specs. Once the CR is deleted,
+// the operator will delete all tinkerbell resources.
+type Tinkerbell struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec describes the desired tinkerbell stack state.
+	Spec TinkerbellStackSpec `json:"spec,omitempty"`
+}
+
+// TinkerbellStackSpec specifies details of an Tinkerbell setup.
+type TinkerbellStackSpec struct {
+	// Version is the Tinkerbell CRD version.
+	Version string `json:"version"`
+
+	// ClusterDNS is the IP address of the cluster DNS resolver.
+	ClusterDNS string `json:"clusterDNS"`
```

Comment:
what is this going to be use for? I know we talked about it, but anyone outside that conversation won't know. It currently doesnt exist anywhere in the helm chart if im not mistaken.

Reply:
So it is actually set here: https://github.com/tinkerbell/operator/blob/main/pkg/resources/tink/configmap.go#L24 and for example used here https://github.com/tinkerbell/operator/blob/main/pkg/resources/tink/configmap.go#L67. The main reason for that is, to configure the nginx dns resolving from the operator CRD instead of relying on reading the the dns nameserver from within the pod as we do in the charts(reading the /etc/resolve.conf) 

For reference https://github.com/tinkerbell/charts/pull/47

Reply:
this is good context, thank you. can we get this in as a code comment?

Reply:
Sure thing, actually I am thinking of even renaming to be more explicit, something like TinkStackResolver, WDYT ?